src/main/java/org/apache/sysds/api/jmlc/Connection.java

```diff
@@ -145,23 +146,15 @@ public Connection(DMLConfig dmlconfig) {
 		
 		//setup basic parameters for embedded execution
 		//(parser, compiler, and runtime parameters)
-		CompilerConfig cconf = new CompilerConfig();
-		cconf.set(ConfigType.IGNORE_UNSPECIFIED_ARGS, true);
-		cconf.set(ConfigType.IGNORE_READ_WRITE_METADATA, true);
-		cconf.set(ConfigType.IGNORE_TEMPORARY_FILENAMES, true);
-		cconf.set(ConfigType.REJECT_READ_WRITE_UNKNOWNS, false);
-		cconf.set(ConfigType.PARALLEL_CP_READ_TEXTFORMATS, false);
-		cconf.set(ConfigType.PARALLEL_CP_WRITE_TEXTFORMATS, false);
-		cconf.set(ConfigType.PARALLEL_CP_READ_BINARYFORMATS, false);
-		cconf.set(ConfigType.PARALLEL_CP_WRITE_BINARYFORMATS, false);
-		cconf.set(ConfigType.PARALLEL_CP_MATRIX_OPERATIONS, false);
-		cconf.set(ConfigType.PARALLEL_LOCAL_OR_REMOTE_PARFOR, false);
-		cconf.set(ConfigType.ALLOW_DYN_RECOMPILATION, false);
-		cconf.set(ConfigType.ALLOW_INDIVIDUAL_SB_SPECIFIC_OPS, false);
-		cconf.set(ConfigType.ALLOW_CSE_PERSISTENT_READS, false);
-		cconf.set(ConfigType.CODEGEN_ENABLED, false);
-		_cconf = cconf;
-		
+		_cconf = OptimizerUtils.constructCompilerConfig(dmlconfig);
+		_cconf.set(ConfigType.IGNORE_UNSPECIFIED_ARGS, true);
+		_cconf.set(ConfigType.IGNORE_READ_WRITE_METADATA, true);
+		_cconf.set(ConfigType.IGNORE_TEMPORARY_FILENAMES, true);
+		_cconf.set(ConfigType.REJECT_READ_WRITE_UNKNOWNS, false);
+		_cconf.set(ConfigType.ALLOW_INDIVIDUAL_SB_SPECIFIC_OPS, false);
+		_cconf.set(ConfigType.ALLOW_CSE_PERSISTENT_READS, false);
+		_cconf.set(ConfigType.CODEGEN_ENABLED, false);
```

Comment:
It looks good to me, but i think many of these settings should not be set custom here.

I we can i suggest removing all of them.

Reply:
We need those, otherwise we can't execute the scripts as they read from files that do not exist, but are provided to the runtime by python.

Reply:
but Codegen for instance?
I get the Reject_read_write unknowns should be enabled.